futures-util/Cargo.toml

```diff
@@ -15,8 +15,9 @@ Common utilities and extension traits for the futures-rs library.
 name = "futures_util"
 
 [features]
-std = ["alloc", "futures-core-preview/std", "futures-io-preview/std", "futures-sink-preview/std", "futures-select-macro-preview/std", "rand", "rand_core", "slab"]
+std = ["alloc", "futures-core-preview/std", "futures-io-preview/std", "futures-sink-preview/std", "futures-select-macro-preview/std", "slab"]
 default = ["std"]
+async-await = ["rand", "rand_core"]
```

Comment:
You can also make `futures-select-macro-preview`, `proc-macro-hack`, and `proc-macro-nested` optional based on the `async-await` feature.

Reply:
`futures-select-macro-preview` has the `futures-select-macro-preview/std` feature and Cargo doesn't really support optionally activating features of optional dependencies, but since that's currently unused I can just remove it and make these optional.

Reply:
Yeah, I don't think `futures-select-macro-preview` makes any sense without `std`-- I think the feature is just there to match other libs. Certainly `select!` doesn't work on no_std today because `await!` doesn't work on `no_std` today.